`sh: shell: command not found` when installing assets

```
$ php bakery build-assets
UserFrosting's Assets Builder
=============================

Installing npm dependencies
---------------------------

> npm install
Skipping as package-lock.json age indicates dependencies are already installed

Installing frontend vendor assets
---------------------------------

> npm run uf-assets-install

> @ uf-assets-install /Users/malou/Desktop/UserFrosting/build
> shell './node_modules/.bin/gulp assetsInstall'

sh: shell: command not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! @ uf-assets-install: `shell './node_modules/.bin/gulp assetsInstall'`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the @ uf-assets-install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Reproduced locally using PHP 7.4.16 on MacOS (node v12.18.3) and Ubuntu VM (v14.17.3).

Running `./node_modules/.bin/gulp assetsInstall` from `build/` works fine as a workaround. 

I suspect an interference from PHP passthrough.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,7 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## [Unknown]
+## [v4.5.1]
+
+### Fixed
+- Fix `php bakery route:list` error on procedural routes ([#1162](https://github.com/userfrosting/UserFrosting/issues/1162)).
+- Fix `NO_DATA` alert when editing a User Role ([#1163](https://github.com/userfrosting/UserFrosting/issues/1163)).
+- [Vagrant/Homestead] Force use of PHP 7.4 for CLI (since default is now PHP 8).
 
 ### Changed
 - Updated Docker development images (PHP 7.2 to 7.4, NodeJS 12.x to 14.x) ([#1085]).
@@ -1069,3 +1074,4 @@ See [http://learn.userfrosting.com/upgrading/40-to-41](Upgrading 4.0.x to 4.1.x 
 [v4.4.4]: https://github.com/userfrosting/UserFrosting/compare/v4.4.3...v4.4.4
 [v4.4.5]: https://github.com/userfrosting/UserFrosting/compare/v4.4.4...v4.4.5
 [v4.5.0]: https://github.com/userfrosting/UserFrosting/compare/v4.4.5...v4.5.0
+[v4.5.1]: https://github.com/userfrosting/UserFrosting/compare/v4.5.0...v4.5.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fix `php bakery route:list` error on procedural routes ([#1162](https://github.com/userfrosting/UserFrosting/issues/1162)).
 - Fix `NO_DATA` alert when editing a User Role ([#1163](https://github.com/userfrosting/UserFrosting/issues/1163)).
 - [Vagrant/Homestead] Force use of PHP 7.4 for CLI (since default is now PHP 8).
+- Fix integration with [filp/whoops 2.14](https://github.com/filp/whoops/compare/2.13.0...2.14.0)
 
 ### Changed
 - Updated Docker development images (PHP 7.2 to 7.4, NodeJS 12.x to 14.x) ([#1085]).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [Unknown]
+
+### Changed
+- Updated Docker development images (PHP 7.2 to 7.4, NodeJS 12.x to 14.x) ([#1085]).
+
 ## [v4.5.0]
 
 ### Changed Requirements
@@ -1047,6 +1052,7 @@ See [http://learn.userfrosting.com/upgrading/40-to-41](Upgrading 4.0.x to 4.1.x 
 [#1121]: https://github.com/userfrosting/UserFrosting/pull/1121
 [#1077]: https://github.com/userfrosting/UserFrosting/pull/1077
 [#1133]: https://github.com/userfrosting/UserFrosting/issues/1133
+[#1085]: https://github.com/userfrosting/UserFrosting/pull/1085
 
 [v4.2.0]: https://github.com/userfrosting/UserFrosting/compare/v4.1.22...v4.2.0
 [v4.2.1]: https://github.com/userfrosting/UserFrosting/compare/v4.2.0...v.4.2.1

--- a/app/defines.php
+++ b/app/defines.php
@@ -11,7 +11,7 @@
 namespace UserFrosting;
 
 // Some standard defines
-define('UserFrosting\VERSION', '4.5.0');
+define('UserFrosting\VERSION', '4.5.1');
 define('UserFrosting\DS', '/');
 define('UserFrosting\PHP_MIN_VERSION', '^7.2');
 define('UserFrosting\PHP_RECOMMENDED_VERSION', '^7.4');

--- a/app/sprinkles/admin/templates/modals/user-manage-roles.html.twig
+++ b/app/sprinkles/admin/templates/modals/user-manage-roles.html.twig
@@ -58,7 +58,7 @@
     <tr class="uf-collection-row">
         <td>
             {{name}}
-            <input type="hidden" name="value[{{ rownum }}][role_id]" value="{{id}}">
+            <input type="hidden" name="roles[{{ rownum }}][role_id]" value="{{id}}">
         </td>
         <td>
             {{description}}

--- a/app/sprinkles/core/composer.json
+++ b/app/sprinkles/core/composer.json
@@ -29,7 +29,7 @@
     "require": {
         "composer/semver": "^3.2.4",
         "doctrine/dbal": "^2.5",
-        "filp/whoops": "^2.3.1",
+        "filp/whoops": "^2.14.0",
         "illuminate/cache": "5.8.*",
         "illuminate/database": "5.8.*",
         "illuminate/events": "5.8.*",

--- a/app/sprinkles/core/src/Bakery/RouteListCommand.php
+++ b/app/sprinkles/core/src/Bakery/RouteListCommand.php
@@ -92,7 +92,7 @@ class RouteListCommand extends BaseCommand
     protected function getRouteInformation(Route $route, InputInterface $input)
     {
         $callable = is_string($route->getCallable()) ? $route->getCallable() : 'Callable';
-        
+
         return $this->filterRoute([
             'method' => implode('|', $route->getMethods()),
             'uri'    => $route->getPattern(),

--- a/app/sprinkles/core/src/Bakery/RouteListCommand.php
+++ b/app/sprinkles/core/src/Bakery/RouteListCommand.php
@@ -78,7 +78,7 @@ class RouteListCommand extends BaseCommand
         }
 
         // Display routes
-        $this->io->table($this->headers, array_filter($routes));
+        $this->io->table($this->headers, $routes);
     }
 
     /**
@@ -91,11 +91,13 @@ class RouteListCommand extends BaseCommand
      */
     protected function getRouteInformation(Route $route, InputInterface $input)
     {
+        $callable = is_string($route->getCallable()) ? $route->getCallable() : 'Callable';
+        
         return $this->filterRoute([
             'method' => implode('|', $route->getMethods()),
             'uri'    => $route->getPattern(),
             'name'   => $route->getName(),
-            'action' => $route->getCallable(),
+            'action' => $callable,
         ], $input);
     }
 

--- a/app/sprinkles/core/src/Error/Renderer/WhoopsRenderer.php
+++ b/app/sprinkles/core/src/Error/Renderer/WhoopsRenderer.php
@@ -192,7 +192,8 @@ class WhoopsRenderer extends ErrorRenderer
         $templateFile = $this->getResource('views/layout.html.php');
         $cssFile = $this->getResource('css/whoops.base.css');
         $zeptoFile = $this->getResource('js/zepto.min.js');
-        $prettifyFile = $this->getResource('js/prettify.min.js');
+        $prismJs = $this->getResource('js/prism.js');
+        $prismCss = $this->getResource('css/prism.css');
         $clipboard = $this->getResource('js/clipboard.min.js');
         $jsFile = $this->getResource('js/whoops.base.js');
 
@@ -234,7 +235,8 @@ class WhoopsRenderer extends ErrorRenderer
             // @todo: Asset compiler
             'stylesheet' => file_get_contents($cssFile),
             'zepto'      => file_get_contents($zeptoFile),
-            'prettify'   => file_get_contents($prettifyFile),
+            'prismJs'    => file_get_contents($prismJs),
+            'prismCss'   => file_get_contents($prismCss),
             'clipboard'  => file_get_contents($clipboard),
             'javascript' => file_get_contents($jsFile),
 

--- a/docker/app/Dockerfile
+++ b/docker/app/Dockerfile
@@ -1,11 +1,11 @@
-FROM php:7.2-fpm
+FROM php:7.4-fpm
 
 RUN apt-get update
 
 RUN apt-get install -y apt-utils
 
 # Add NodeJS LTS Repository
-RUN curl -sL https://deb.nodesource.com/setup_12.x | bash -
+RUN curl -sL https://deb.nodesource.com/setup_14.x | bash -
 
 RUN apt-get install -y \
     libc-client-dev \
@@ -16,20 +16,35 @@ RUN apt-get install -y \
     libxml2-dev \
     locales \
     zlib1g-dev \
+    libzip-dev\
     libpng-dev \
     libfreetype6-dev \
+    libjpeg62-turbo-dev \
     libjpeg-dev \
     zip \
     nodejs
 
+# Optional XDebug Install
+#RUN curl -fsSL 'https://xdebug.org/files/xdebug-2.9.4.tgz' -o xdebug.tar.gz \
+#    && mkdir -p xdebug \
+#    && tar -xf xdebug.tar.gz -C xdebug --strip-components=1 \
+#    && rm xdebug.tar.gz \
+#    && ( \
+#    cd xdebug \
+#    && phpize \
+#    && ./configure --enable-xdebug \
+#    && make -j$(nproc) \
+#    && make install \
+#    ) \
+#    && rm -r xdebug \
+#    && docker-php-ext-enable xdebug   
+
 # COMPOSER INSTALL
 RUN curl -sSfo /tmp/composer.phar https://getcomposer.org/installer
 RUN php /tmp/composer.phar --install-dir=/usr/local/bin --filename=composer
-# Force Composer Stable Version 1
-RUN composer selfupdate --1
 
 # Install and configure PHP extensions
-RUN docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-jpeg-dir=/usr/include/ \
+RUN docker-php-ext-configure gd --with-freetype --with-jpeg \
     && docker-php-ext-install -j$(nproc) gd \
     && docker-php-ext-install -j$(nproc) pdo pdo_mysql \
     && docker-php-ext-install -j$(nproc) zip

--- a/vagrant/after.sh
+++ b/vagrant/after.sh
@@ -14,6 +14,10 @@ npm cache clean -f
 npm install -g n
 n -q lts
 
+# Ensuite PHP 7.4 is used (UF doesn't support PHP 8 yet...)
+echo "\n\n >> Forcing PHP 7.4 in CLI\n"
+sudo update-alternatives --set php /usr/bin/php7.4
+
 # Ensure composer deps are installed
 echo "\n\n >> Installating Composer\n"
 cd ${BASE_PATH}

